src/Microsoft.AspNet.Authentication/AuthenticationOptions.cs

```diff
@@ -42,5 +43,11 @@ public string AuthenticationScheme
         /// Additional information about the authentication type which is made available to the application.
         /// </summary>
         public AuthenticationDescription Description { get; set; } = new AuthenticationDescription();
+
+        /// <summary>
+        /// The request path within the application's base path where the user-agent will be returned
+        /// when a failure occurs.
+        /// </summary>
+        public PathString ErrorHandlerPath { get; set; }
```

Comment:
Yeah well the other option is to introduce a base class for all of the Invoke style middlewares (not cookie/bearer) basically.  I've been leaning towards that anyways, I'm not opposed to taking a stab at that and moving this error handling into the new common base.  We need to rename InvokeAsync anyways, that name is terrible.